src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -75,44 +64,43 @@ end
 -- main import functions (split up to allow for Wait conditions)
 ---------------------------------------------------------
 
--- Identifies import token, determines campaign box and downloads it (if needed)
-function findCampaignFromToken(_, _, _)
-  local coin = nil
-  local coinObjects = getObjectsWithTag("ImporterToken")
-  if #coinObjects == 0 then
-    broadcastToAll("Could not find importer token", Color.Red)
-  elseif #coinObjects > 1 then
-    broadcastToAll("More than 1 importer token found. Please delete all but 1 importer token", Color.Yellow)
-  else
-    coin = coinObjects[1]
+function onCollisionEnter(info)
+  local objectData = info.collision_object.getData()
+  if info.collision_object.hasTag("ImporterToken") then
+    findCampaignFromToken(info.collision_object)
+  end
+end
 
-    local importData = JSON.decode(coin.getGMNotes())
-    campaignBoxGUID = importData["box"]
+-- Identifies import token, determines campaign box and downloads it (if needed)
+function findCampaignFromToken(coin)
```

Comment:
I'd give this function a better fitting name. Something like `ImportFromToken()`?